tests/integration_tests/scheduler/test_lsf_driver.py

```diff
@@ -217,3 +218,47 @@ def mock_poll_once_by_bhist(*args, **kwargs):
     assert "bhist is used" in caplog.text
     assert bhist_called
     assert job_id in driver._bhist_cache
+
+
+async def test_kill_before_submit_is_finished(tmp_path, monkeypatch, caplog):
+    print("")
+    os.chdir(tmp_path)
+
+    bin_path = tmp_path / "bin"
+    bin_path.mkdir()
+    monkeypatch.setenv("PATH", f"{bin_path}:{os.environ['PATH']}")
+    bsub_path = bin_path / "slow_bsub"
+    bsub_path.write_text(
+        "#!/bin/sh\nsleep 0.05\nbsub $@",
+        encoding="utf-8",
+    )
+    bsub_path.chmod(bsub_path.stat().st_mode | stat.S_IEXEC)
+
+    caplog.set_level(logging.DEBUG)
+    driver = LsfDriver(bsub_cmd="slow_bsub")
+
+    # Allow submit and kill to be interleaved by asyncio by issuing
+    # submit() in its own asyncio Task:
+    asyncio.create_task(
+        driver.submit(
+            # The sleep is the time window in which we can kill the job before
+            # the unwanted finish message appears on disk.
+            0,
+            "sh",
+            "-c",
+            f"sleep 1; touch {tmp_path}/survived",
```

Comment:
Shall we sleep (eg 5 secs) more here? Just to remove potential flakiness....

Reply:
Added some if's to guard against flakyness. This test is also relevant in interactive pytest runs so then I don't want to sleep for too long.